File: Various/GCD.py
def gcd(a, b):                   # greatest common divisor of a,b
   if b != 0:                    # Recursive implementation of Euclid's
      return gcd(b, a%b);
   return a


def inverse(a, m):                     # Solves a*x = 1 (mod m)
   if gcd(a,m) != 1:                   # Recursive application of Euclid's
      print 'Not relatively prime';    # Short-short code from Bruce
      return
   a = a%m;
   if a == 1:
      return 1;
   return ( (1-inverse(m,a)*m)/a )%m

File: Various/GCD.cpp
#include <iostream>

#define ll long long 

using namespace std;

int gcd(int a, int  b)
{
   if (b == 0)
      return a;
   return gcd(b, a%b);
}


int inverse(int a, int m)           // Inverse of a mod m
{
//   cout << "Entering inverse() " << a << m << endl;
   if (gcd(a,m) != 1)
      return -1;
   a = a%m;
   if (a == 1)
      return 1;
   return ( (1 - inverse(m,a)*m)/a ) % m;

}


int main()
{
   int a, b;
   cout << "First number: ";
   cin >> a;
   cout << "Second number: ";
   cin >> b;
   cout << gcd(a,b) << endl;
r//   cout << a%b << endl;
   cout << inverse(a,b) << endl;
   return 0;
}
